galaxy_ng/app/migrations/0047_update_role_search_vector_trigger.py

```diff
@@ -0,0 +1,64 @@
+# Generated by Django 4.2.7 on 2023-11-15 15:55
+
+from django.db import migrations
+
+REBUILD_ROLES_TS_VECTOR = """
+UPDATE galaxy_legacyrole SET name = name;
+"""
+
+CREATE_ROLE_TS_VECTOR_TRIGGER = """
+CREATE OR REPLACE FUNCTION update_role_ts_vector()
+    RETURNS TRIGGER
+    AS $$
+DECLARE
+    _search_vector tsvector;
+    _namespace text;
+BEGIN
+    _namespace := (select name from galaxy_legacynamespace WHERE id = NEW.namespace_id);
+    _search_vector := ((((
+        setweight(to_tsvector(COALESCE(_namespace, '')), 'A')
+        || setweight(to_tsvector(COALESCE(NEW."name", '')), 'A'))
+        || setweight(to_tsvector(COALESCE(((NEW."full_metadata"->'tags'))::text, '')), 'B')) 
+        || setweight(to_tsvector(COALESCE(((NEW."full_metadata"->'platforms'))::text, '')), 'C')) 
+        || setweight(to_tsvector(COALESCE((NEW."full_metadata"->>'description'), '')), 'D'));
+
+    INSERT INTO galaxy_legacyrolesearchvector(role_id,search_vector,modified)
+        VALUES(new.id,_search_vector,current_timestamp)
+    ON CONFLICT (role_id)
+        DO UPDATE SET
+            search_vector = _search_vector, modified = current_timestamp;
+    RETURN NEW;
+END;
+$$
+LANGUAGE plpgsql;
+
+DROP TRIGGER IF EXISTS update_ts_vector ON galaxy_legacyrole;
+
+CREATE TRIGGER update_ts_vector
+    AFTER INSERT OR UPDATE
+    ON galaxy_legacyrole
+    FOR EACH ROW
+EXECUTE PROCEDURE update_role_ts_vector();
```

Comment:
```suggestion
EXECUTE FUNCTION update_role_ts_vector();
```

Reply:
Is there any advantage on using FUNCTION instead of PROCEDURE? or is it just to not be confusing with stored procedures?

Reply:
In your case both PROCEDURE or FUNCTION will work as expected. The difference is in Postgres PROCEDUREs you can do transaction control inside (aka COMMIT/ROLLBACK). But in this case I guess you want to be full atomic so just to don't confusing people because you created a function and then in the trigger definition you're referencing a procedure.